ykrt/src/compile/jitc_yk/codegen/x86_64/deopt.rs

```diff
@@ -24,6 +24,48 @@ pub(crate) extern "C" fn __yk_deopt(
     let aot_smaps = AOT_STACKMAPS.as_ref().unwrap();
     let info = &ctr.deoptinfo[deoptid];
 
+    if let Some(st) = info.guard.getct() {
+        // Prepare the traceinputs "struct" (for now this is just a vector) and pass it into the
+        // side-trace.
+        let mut ykctrlpvars = Vec::new();
+        for jitval in &info.lives {
+            let val = match jitval {
+                LocalAlloc::Stack { frame_off, size } => {
+                    let p = unsafe { jitrbp.byte_sub(*frame_off) };
+                    match *size {
+                        1 => unsafe { u64::from(std::ptr::read::<u8>(p as *const u8)) },
+                        2 => unsafe { u64::from(std::ptr::read::<u16>(p as *const u16)) },
+                        4 => unsafe { u64::from(std::ptr::read::<u32>(p as *const u32)) },
+                        8 => unsafe { std::ptr::read::<u64>(p as *const u64) },
+                        _ => todo!(),
+                    }
+                }
+                LocalAlloc::Register => todo!(),
+                LocalAlloc::ConstFloat(_) => todo!(),
+                LocalAlloc::ConstInt(c) => *c,
```

Comment:
This enum variant has been slightly changed (but not in a very difficult way) by #1285 so you might want to merge master in,

Reply:
Ah this merged after I raised this PR. Will fix.

Reply:
[5d1865c](https://github.com/ykjit/yk/pull/1287/commits/5d1865c361e3c94685c54bd257f4d2ef052de889)